### PR TITLE
Use VerifyEmailViewMixin for the UI version of the view

### PR DESCRIPTION
**Pull request by adam-thomas**
@incuna/backend review please?

### CODE REVIEW

**Comment by adam-thomas** [comment]
bump?

**Comment by adam-thomas** [comment]
guise plz

File: user_management/ui/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from django.http import Http404
+from django.utils.translation import ugettext_lazy as _
+
+
+class InvalidExpiredToken(Http404):
+    """Exception to confirm an account."""
+    message = _('Invalid or expired token.')
```
Comment (Ian-Foote) [inline user_management/ui/exceptions.py]: I really don't think this should be a `404`. It looks more like a `401` or maybe a `403` to me.

Comment (adam-thomas) [inline user_management/ui/exceptions.py]: The API version returns a `400`.  Unfortunately I can't find a way to generate a HTTP400 exception from a normal Django site - any ideas?

Comment (Ian-Foote) [inline user_management/ui/exceptions.py]: Return a [`HttpResponseBadRequest`](https://docs.djangoproject.com/en/1.9/ref/request-response/#django.http.HttpResponseBadRequest)?

Comment (adam-thomas) [inline user_management/ui/exceptions.py]: That's a reasonably big refactor, mind if I do it in a separate PR?

**Comment by adam-thomas** [comment]
Any more comments for this particular PR?

**Comment by adam-thomas** [comment]
@incuna/backend further bumpage (sorry @meshy) :P

**Comment by KevinEtienne** [comment]
I was waiting for a response to https://github.com/adam-thomas but looks good to me

**Comment by adam-thomas** [comment]
Yeah that's why I was bumping the thread :P  Cheers, I can come back to it if there are any further issues.

File: user_management/ui/exceptions.py (hunk 1 above)
Comment (Ian-Foote) [inline user_management/ui/exceptions.py]: Can you explain why?

Comment (adam-thomas) [inline user_management/ui/exceptions.py]: As I understand it, Kevin's logic was that the `400` response is actually used and displayed by a frontend, whereas this error isn't handled by anything so we want the browser to show up a `404`.

Technically, I guess it's also more correct - the request is fine, it's the URL that's wrong, since that's where the token is delivered.  That's, like, maximal pedant, though.
